### PR TITLE
Fix Error in Defining Particle Filters in EAGLE

### PR DESCRIPTION
**Pull request by stephaniehho**
Particle Filter cannot be set while using the EAGLE simulation.  It traces to the error of 'UnboundLocalError: local variable 'ftype' referenced before assignment'  at the following location: yt/frontends/owls/fields.pyc in setup_particle_fields(self, ptype).

This can be solved by adding an else-case, which set ftype=ptype.  This change is submitted to this pull request.

## PR Checklist

- [ ] Code passes flake8 checker
- [ ] New features are documented, with docstrings and narrative docs
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: yt/frontends/owls/fields.py [hunk 1]
```diff
@@ -108,6 +108,9 @@ def setup_particle_fields(self, ptype):
             ftype='BH'
         elif ptype == 'all':
```
Comment (ngoldbaum) [inline yt/frontends/owls/fields.py]: You can delete this elif clause too

File: yt/frontends/owls/tests/test_outputs.py [hunk 2]
```diff
@@ -54,3 +55,23 @@ def test_snapshot_033():
...
+    ad = ds.all_data()
+    def cold_gas(pfilter, data):
+        temperature = data[pfilter.filtered_type, "Temperature"]
+        filter = (temperature.in_units('K') <= 1e5)
+        return filter
+    add_particle_filter("gas_cold", function=cold_gas, filtered_type='PartType0', requires=["Temperature"])
+    ds.add_particle_filter('gas_cold')
+
+    mask = (ad['PartType0','Temperature'] <= 1e5)
+    try:
```
Comment (ngoldbaum) [inline yt/frontends/owls/tests/test_outputs.py]: I would replace this whole try/except block with the following line:

```
assert ad['PartType0','Temperature'][mask].shape == ad['gas_cold','Temperature'].shape
```

There's no need for fancy error handling in a test, if the test breaks we'd like to know where it breaks :)

**Comment by ngoldbaum** [comment]
@yt-fido test this please

File: yt/frontends/owls/tests/test_outputs.py [hunk 3]
```diff
@@ -54,3 +55,19 @@ def test_snapshot_033():
 @requires_file(os33)
 def test_OWLSDataset():
     assert isinstance(data_dir_load(os33), OWLSDataset)
+    
+    
+@requires_ds(os33)
+def test_OWLS_particlefilter(os33):
```
Comment (ngoldbaum) [inline yt/frontends/owls/tests/test_outputs.py]: This function shouldn't take any arguments, this should look like:

```
def test_OWLS_particlefilter():
```

That should fix the test failure.

**Comment by stephaniehho** [comment]
Sorry about that error.  Just made the change.

**Pull request merged**